streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java

```diff
@@ -70,6 +72,45 @@
     }
 
     @Test
+    public void shouldOnlyIterateOpenSegments() throws Exception {
+        final File baseDir = TestUtils.tempDir();
+        Producer<byte[], byte[]> producer = new MockProducer<>(true, byteArraySerde.serializer(), byteArraySerde.serializer());
+        RecordCollector recordCollector = new RecordCollector(producer) {
+            @Override
+            public <K1, V1> void send(ProducerRecord<K1, V1> record, Serializer<K1> keySerializer, Serializer<V1> valueSerializer) {
+            }
+        };
+
+        MockProcessorContext context = new MockProcessorContext(
+                null, baseDir,
+                byteArraySerde, byteArraySerde,
+                recordCollector);
+
+        final WindowStore<Integer, String> windowStore = createWindowStore(context);
+        long currentTime = 0;
+        context.setTime(currentTime);
+        windowStore.put(1, "one");
+        currentTime = currentTime + segmentSize;
+        context.setTime(currentTime);
+        windowStore.put(1, "two");
+        currentTime = currentTime + segmentSize;
+        context.setTime(currentTime);
+        windowStore.put(1, "three");
+
+        final WindowStoreIterator<String> iterator = windowStore.fetch(1, 0, currentTime);
+
+        // roll to the next segment that will close the first
+        currentTime = currentTime + segmentSize;
+        context.setTime(currentTime);
+        windowStore.put(1, "four");
+
+        // should only have 2 values as the first segment is no longer open
+        assertEquals(new KeyValue<>(60000L, "two"), iterator.next());
```

Comment:
Yep - `segmentSize` in the test is initialized as `RocksDBWindowStoreTest#MIN_SEGMENT_INTERVAL`